crates/control/src/role_assignment.rs

```diff
@@ -268,17 +276,40 @@ impl RoleAssignment {
                 );
             }
         }
-
-        if let Some(last_time_keeper_penalized) = self.last_time_keeper_penalized {
-            let deny_replacement_keeper_switch = cycle_start_time
-                .duration_since(last_time_keeper_penalized)
-                .expect("Keeper was penalized in the Future")
-                < *context.keeper_replacementkeeper_switch_time;
-            if self.role == Role::ReplacementKeeper
-                && !send_spl_striker_message
-                && deny_replacement_keeper_switch
-            {
-                role = Role::ReplacementKeeper;
+        if self.role == Role::ReplacementKeeper {
+            if let Some(last_time_keeper_penalized) = self.last_time_player_was_penalized.one {
```

Comment:
Maybe I dont understand this completely. It looks to me as if you could merge the two match arms since they essentially do the same.

Reply:
You were right, this makes this a lot shorter